### PR TITLE
Chipyard-based Toolchain

### PR DESCRIPTION
**Pull request by alonamid**
Re-structuring of the riscv toolchain component of build-setup.sh to play nicely with the Chipyard toolchain setup.

Depends on Chipyard PR 157 (https://github.com/ucb-bar/chipyard/pull/157)

- [x] Update the pre-compiled toolchain to include rocket-tools hash instead of riscv-tools
- [x] Make decisions regarding fesvr getting pulled out of riscv-tools/rocket-tools into spike

### CODE REVIEW

File: build-setup-nolog.sh [hunk 1]
```diff
@@ -65,15 +65,15 @@ done
 # ignore riscv-tools for submodule init recursive
 # you must do this globally (otherwise riscv-tools deep
 # in the submodule tree will get pulled anyway
-git config --global submodule.riscv-tools.update none
+git config --global submodule.toolchains/riscv-tools.update none
```
Comment (davidbiancolin) [inline build-setup-nolog.sh]: consider dropping the `--global`

File: build-setup-nolog.sh [hunk 2]
```diff
@@ -97,82 +97,34 @@ fi
 #   the user to rerun this script without --fast
 # 2) If fast was not specified, but the toolchain from source
 if [ "$IS_LIBRARY" = true ]; then
-    target_toolchain_dir=$RDIR/../../toolchains/
+    target_toolchain_dir=$RDIR/../../toolchains
```
Comment (davidbiancolin) [inline build-setup-nolog.sh]: I think this is more of a nit, but the rest of the code is cleaned up if you make this a "chipyard_dir".

File: build-setup-nolog.sh [hunk 3]
```diff
@@ -97,82 +97,34 @@ fi
...
-    echo "export FIRESIM_STANDALONE=1" >> env.sh
+  $target_toolchain_dir/../scripts/build-toolchains.sh
 fi
+cd $RDIR
+#generate env.sh file which sources the chipyard env.sh file
+echo "if [ -f \"$target_toolchain_dir/../env.sh\" ]; then" > env.sh
+echo "  source $target_toolchain_dir/../env.sh" >> env.sh
+echo "  export FIRESIM_ENV_SOURCED=1" >> env.sh
+echo "else" >> env.sh
+echo "  echo \"Error: You may have forgot to build or source the toolchains (build them independantly in firesim-as-a-library mode)\"" >> env.sh
```
Comment (davidbiancolin) [inline build-setup-nolog.sh]: independently

File: build-setup-nolog.sh [hunk 4]
```diff
@@ -97,82 +97,34 @@ fi
...
-
-if  [ "$IS_LIBRARY" = false ]; then
-    echo "export FIRESIM_STANDALONE=1" >> env.sh
+  $target_toolchain_dir/../scripts/build-toolchains.sh
 fi
+cd $RDIR
+#generate env.sh file which sources the chipyard env.sh file
+echo "if [ -f \"$target_toolchain_dir/../env.sh\" ]; then" > env.sh
+echo "  source $target_toolchain_dir/../env.sh" >> env.sh
+echo "  export FIRESIM_ENV_SOURCED=1" >> env.sh
```
Comment (davidbiancolin) [inline build-setup-nolog.sh]: where is the "FIRESIM_STANDALONE" ?

File: build-setup-nolog.sh (hunk 1 above)
Comment (davidbiancolin) [reply build-setup-nolog.sh]: (In both places)

Comment (alonamid) [reply build-setup-nolog.sh]: fixed

File: build-setup-nolog.sh (hunk 2 above)
Comment (alonamid) [reply build-setup-nolog.sh]: I agree. fixed

File: build-setup-nolog.sh (hunk 3 above)
Comment (alonamid) [reply build-setup-nolog.sh]: Fixed

File: build-setup-nolog.sh (hunk 4 above)
Comment (alonamid) [reply build-setup-nolog.sh]: I lost it somewhere in the way of deleting the toolchain build section :).
Fixed

**Comment by davidbiancolin** [comment]
It feels a little weird, but since this PR is bumping chipyard you should regenerate AGFIs.

**Comment by alonamid** [comment]
Do we need to re-generate agif's on dev?

**Pull request merged**